doc: DRAFT template for 1.12 release notes

dbkinder opened the pull request:
Get the 1.12 release notes started with material derived from the 1.12
roadmap and headings from the 1.11 release notes.

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

henrikbrixandersen committed:
samples: bluetooth: Add Apple iBeacon demo application

Add a simple demo implementation of an Apple iBeacon BLE broadcaster.

Signed-off-by: Henrik Brix Andersen <henrik@brixandersen.dk>

henrikbrixandersen committed:
samples: flash_shell: Use correct flash write block size

Change the flash_shell sample from using FLASH_WRITE_BLOCK_SIZE
definition to using the flash_get_write_block_size() api for the
selected flash_device.

This fixes the flash_shell sample when using multiple flash devices
with different write block sizes.

Signed-off-by: Henrik Brix Andersen <henrik@brixandersen.dk>

spoorthik committed:
tests: threads: Add test to verify delayed thread abort

As k_thread_cancel() is deprecated, we need to test if delayed thread
which is in wait queue can be cancelled from k_thread_abort().

Signed-off-by: Spoorthi K <spoorthi.k@intel.com>

dbkinder committed:
doc: DRAFT template for 1.12 release notes

Get the 1.12 release notes started with material derived from the 1.12
roadmap and headings from the 1.11 release notes.

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

dbkinder committed:
doc: DRAFT template for 1.12 release notes

Get the 1.12 release notes started with material derived from the 1.12
roadmap and headings from the 1.11 release notes.

Signed-off-by: David B. Kinder <david.b.kinder@intel.com>

kestewart commented:
I don't think we want to declare this one as LTS,  but rather wait until the MISRA-C work is done. 
+* First LTS release, with QM (Quality Managed) level qualification

dbkinder commented:
OK, I just removed that line for now.  There will be tweaks to the others as we get closer to the release but I wanted to start with something based on the roadmap.

rgundi committed:
kernel: sem: Modify the way BUILD_ASSERT is used

BUILD_ASSERT() macro makes use of __COUNTER__ which may not be
supported in some compilers (like xcc). So, multiple uses of
BUILD_ASSERT() in same scope is not possible for such compilers.
Instead, the expression to BUILD_ASSERT can be "&&"ed to achieve
the same purpose.

Signed-off-by: Rajavardhan Gundi <rajavardhan.gundi@intel.com>

rgundi committed:
Revert "warnings: Disable "unused-local-typedefs" compiler warning"

This reverts commit d9ac1d4b4a131b354f3600d6a0f14fc4fc01be1a.

This option can be compiler specific and is not required anymore.

Signed-off-by: Rajavardhan Gundi <rajavardhan.gundi@intel.com>
Signed-off-by: Anas Nashif <anas.nashif@intel.com>

Commit:
slist: abstract node and list implementation

Singly-linked lists work in a specific way, but this will allow
slist semantics to be defined for any data structure that can store
a "next" pointer in any way at all, possibly something not even
a pointer value, but a table index.

The immediate need for this patch is to allow the easy definition
of an slist variant which stores flags in the low-order bits of the
node pointer.

The compiler does not need them, but the function prototypes have
been left in slist.h to make the header easier to understand by
the end user and not confuse Doxygen.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

Mani-Sadhasivam committed:
CODEOWNERS: Add myself as the Codeowner for LED API and drivers

Add myself as the Codeowner for LED API and its drivers.

Signed-off-by: Manivannan Sadhasivam <manivannan.sadhasivam@linaro.org>

dbkinder committed:
Merge remote-tracking branch 'upstream/master'

Mani-Sadhasivam committed:
arm: Fix title for SoC configuration in Kconfig

The title for selecting the configuration options for SoC's should be
"SoC Configuration" not "Board Configuration".

Signed-off-by: Manivannan Sadhasivam <manivannan.sadhasivam@linaro.org>

Mani-Sadhasivam committed:
arm: soc: Elaborate Kconfig strings for MPU selection

Some of the SoCs specify "Enable MPU" in Kconfig for enabling MPU
support. This will create ambiguity while selecting options via
menuconfig. Hence, append SoC family name to the selection strings
to looks more elaborative.

Signed-off-by: Manivannan Sadhasivam <manivannan.sadhasivam@linaro.org>

jukkar committed:
net: arp: Do not access NULL network packet

The code was accessing network pkt before the value of the pkt
was checked.

Coverity-CID: 185394

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

ydamigos committed:
arch/arm/soc/st_stm32: Fix typos in soc.h

Fix typos in soc.h for stm32f1 and stm32l4 families.

Signed-off-by: Yannis Damigos <giannis.damigos@gmail.com>

Commit:
lib: posix: pthread_common: Fix potential integer overflow issue

Fix potential overflow of interger expression for by fixing
variable type to s64_t.

CID: 185275

Signed-off-by: Ramakrishna Pallala <ramakrishna.pallala@intel.com>

Commit:
drivers/spi: Remove DW spi slave test left over

This reset to 0 was used for testing threshold on slave mode.

Coverity-CID: 185402

Signed-off-by: Tomasz Bursztyka <tomasz.bursztyka@linux.intel.com>

nashif committed:
toolchain: update xtools config

Add placeholders for all possible xtools toolchains.

Fixes #6911

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

nashif committed:
build: simplify git describe call

Instead of using a module that tries to do too many things that break
in different places, keep it simple and just call git describe.

This fixes #7044 and fixes #7207.

Signed-off-by: Anas Nashif <anas.nashif@intel.com>

SebastianBoe committed:
kconfig: Move CPLUSPLUS from root to "Compiler Options"

Enabling C++ support for the application has been inappropriately
located at the root of the Kconfig menu. The root should be kept as
clean possible to allow easy navigation.

This commit moves CONFIG_CPLUSPLUS into

~/"Build and Linker Features"/"Compiler Options".

This is a purely cosmetic change and does not change the
'visibility' (depends) of the Kconfig option.

Arguably, it would fit better into

~/"Build and Linker Features"/"Language Options"

but this entry does not exist.

Signed-off-by: Sebastian Bøe <sebastian.boe@nordicsemi.no>

Commit:
kconfig: fix menuconfig

"depends on" cannot use expressions as arguments, this statement
didn't actually do anything for kconfiglib.py and caused an
error with menuconfig.

There's currently no way to depend on a system heap being
present, however forthcoming patches move this feature away
from using the system heap anyway.

For now, if the user enables dynamic objects without defining
a system heap, they will get a build error.

Fixes #7266.

Signed-off-by: Andrew Boie <andrew.p.boie@intel.com>

dbkinder commented:
@MaureenHelm  I think you're the "chosen one" for this release, so if you'd merge this draft release notes doc, others can start filling in their parts.

MaureenHelm reviewed (changes requested):
@dbkinder I appreciate you kick starting this, but I think it will confuse people to highlight new major features that haven't been merged yet (e.g., AMP, multiple git repos, LLVM). Can you remove them for now? Will you please also rebase so we don't have the merge commit?

nashif commented on doc/release-notes-1.12.rst:
```diff
@@ -0,0 +1,80 @@
+
+Major enhancements with this release include:
+
+* Asymmetric multiprocessing (AMP) on supported on multi-core platforms
```
I would prefer if we only added the things that are already merged and closed per the roadmap, not everything to avoid confusion.

dbkinder committed:
Merge branch 'master' of github.com:dbkinder/zephyr

dbkinder commented:
replaced with #7277
